### PR TITLE
Meta link headers

### PR DESCRIPTION
**Pull request by dannylamb**
**GitHub Issue**: Resolves https://github.com/Islandora-CLAW/CLAW/issues/717

# What does this Pull Request do?

Adds `rel=alternate type=mimetype` link headers to GET and HEAD requests, advertising our REST endpoints and what formats are enabled.

# What's new?
- Code to add those headers
- Updated tests

# How should this be tested?

- Make a node
- Perform a HEAD request against its canonical URL (note we're doing this as anonymous)
```bash
ubuntu@claw:/var/www/html/drupal/web/modules/contrib/islandora$ curl -I http://localhost:8000/node/1
HTTP/1.1 200 OK
Date: Mon, 12 Mar 2018 14:48:50 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.25-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/1?_format=jsonld>; rel="alternate"; type="application/ld+json"
Link: <http://localhost:8000/node/1?_format=json>; rel="alternate"; type="application/json"
Link: </node/1>; rel="canonical"
Link: </node/1>; rel="shortlink"
Link: </node/1>; rel="revision"
Link: </node?node=1>; rel="create"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
X-Drupal-Cache: HIT
Content-Type: text/html; charset=UTF-8
```
- Now try it against a REST endpoint.  Note we're authenticated now.  The format you request should not be advertised (e.g. if you're on `?_format=json` then the JSON link header shouldn't be present).
```
ubuntu@claw:/var/www/html/drupal/web/modules/contrib/islandora$ curl -I -u admin:islandora http://localhost:8000/node/1?_format=json
HTTP/1.1 200 OK
Date: Mon, 12 Mar 2018 14:50:57 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.25-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/1>; rel="canonical"
Link: <http://localhost:8000/node/1/delete>; rel="https://drupal.org/link-relations/delete-form"
Link: <http://localhost:8000/node/1/edit>; rel="edit-form"
Link: <http://localhost:8000/node/1/revisions>; rel="version-history"
Link: <http://localhost:8000/node/1>; rel="https://drupal.org/link-relations/revision"
Link: <http://localhost:8000/node?node=1>; rel="https://drupal.org/link-relations/create"
Link: <http://localhost:8000/node/1?_format=jsonld>; rel="alternate"; type="application/ld+json"
X-Drupal-Dynamic-Cache: HIT
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
Content-Type: application/json
```

# Additional Notes:
I think that's enough REST for now? :man_shrugging: 

# Interested parties
@Islandora-CLAW/committers

### CODE REVIEW

**Comment by dannylamb** [comment]
Whoops, I pushed setting up the REST config into the base test class, and missed a few spots to reference the new functionality.

**Comment by dannylamb** [comment]
Sorry 'bout that @Islandora-CLAW/committers, this is ready to test now.

**Comment by mjordan** [comment]
I can test. Question: why do we exclude the mimetype that is requested as opposed to including all the available mimetypes in the response?

**Comment by dannylamb** [comment]
I'm excluding it since the link relationship is 'alternate', implying that there's a second resource.  Maybe it would be appropriate to include itself using `rel="self"`?  So if you visited `localhost:8000/node/1?_format=json` you'd get `Link: <localhost:8000/node/1?_format=json>; rel="self"; type="application/json"` and `Link: <localhost:8000/node/1?_format=jsonld>; rel="alternate"; type="application/ld+json"`.

**Comment by mjordan** [comment]
@dannylamb makes sense. Any clients requesting this information would already know the value of 'self' since they need it to make the request.

**Comment by mjordan** [comment]
Testing this now and am encountering a problem. Here's what's happening:

1. Running the islandora module on 21124f317d14099753104c0e80540bcb8ecf2b17, my site is working fine.
1. After I check out @dannylamb's meta-link-headers branch, my site breaks, leaving the following in the apache error log:

```
[Mon Mar 19 14:01:11.219726 2018] [:error] [pid 2413] [client 10.0.2.2:42913] TypeError: Argument 1 passed to Drupal\\islandora\\EventSubscriber\\LinkHeaderSubscriber::__construct() must be an instance of Drupal\\Core\\Entity\\EntityTypeManager, instance of Drupal\\Core\\Entity\\EntityFieldManager given, called in /var/www/html/drupal/web/core/lib/Drupal/Component/DependencyInjection/Container.php on line 266 in /var/www/html/drupal/web/modules/contrib/islandora/src/EventSubscriber/LinkHeaderSubscriber.php on line 75 #0 /var/www/html/drupal/web/core/lib/Drupal/Component/DependencyInjection/Container.php(266): Drupal\\islandora\\EventSubscriber\\LinkHeaderSubscriber->__construct(Object(Drupal\\Core\\Entity\\EntityFieldManager), Object(Drupal\\Core\\Routing\\CurrentRouteMatch))\n#1 /var/www/html/drupal/web/core/lib/Drupal/Component/DependencyInjection/Container.php(171): Drupal\\Component\\DependencyInjection\\Container->createService(Array, 'islandora.node_...')\n#2 /var/www/html/drupal/web/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php(105): Drupal\\Component\\DependencyInjection\\Container->get('islandora.node_...')\n#3 /var/www/html/drupal/vendor/symfony/http-kernel/HttpKernel.php(191): Drupal\\Component\\EventDispatcher\\ContainerAwareEventDispatcher->dispatch('kernel.response', Object(Symfony\\Component\\HttpKernel\\Event\\FilterResponseEvent))\n#4 /var/www/html/drupal/vendor/symfony/http-kernel/HttpKernel.php(173): Symfony\\Component\\HttpKernel\\HttpKernel->filterResponse(Object(Drupal\\Core\\Render\\HtmlResponse), Object(Symfony\\Component\\HttpFoundation\\Request), 1)\n#5 /var/www/html/drupal/vendor/symfony/http-kernel/HttpKernel.php(68): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)\n#6 /var/www/html/drupal/web/core/lib/Drupal/Core/StackMiddleware/Session.php(57): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#7 /var/www/html/drupal/web/core/lib/Drupal/Core/StackMiddleware/KernelPreHandle.php(47): Drupal\\Core\\StackMiddleware\\Session->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#8 /var/www/html/drupal/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(99): Drupal\\Core\\StackMiddleware\\KernelPreHandle->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#9 /var/www/html/drupal/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(78): Drupal\\page_cache\\StackMiddleware\\PageCache->pass(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#10 /var/www/html/drupal/web/core/lib/Drupal/Core/StackMiddleware/ReverseProxyMiddleware.php(47): Drupal\\page_cache\\StackMiddleware\\PageCache->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#11 /var/www/html/drupal/web/core/lib/Drupal/Core/StackMiddleware/NegotiationMiddleware.php(50): Drupal\\Core\\StackMiddleware\\ReverseProxyMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#12 /var/www/html/drupal/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Drupal\\Core\\StackMiddleware\\NegotiationMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#13 /var/www/html/drupal/web/core/lib/Drupal/Core/DrupalKernel.php(657): Stack\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#14 /var/www/html/drupal/web/index.php(19): Drupal\\Core\\DrupalKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))\n#15 {main}, referer: http://localhost:8000/
```

I've run `drupal router:rebuild` and it makes no difference. Am I missing something obvious?

**Comment by mjordan** [comment]
Updated Islandora to 6e8985728adeb386ecef109898ffed1f870d16df but that didn't help.

**Comment by dannylamb** [comment]
@mjordan let me see what state the services file is in.  either i need to update that or you need to clear your cache.

**Comment by dannylamb** [comment]
@mjordan The new constructor args are in the right order, but your cache will need to get rebuilt to re-parse `islandora.services.yml`.  If `drupal cache:rebuild all` doesn't do it for you, i'll have to dig deeper.  

Probably should've put that in the testing instructions :man_facepalming:

**Comment by mjordan** [comment]
OK, rebuild cache. Here are my results:

```
curl -I http://localhost:8000/node/17
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 18:52:59 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
X-Drupal-Dynamic-Cache: MISS
Link: </node/17>; rel="canonical"
Link: </node/17>; rel="shortlink"
Link: </node/17>; rel="revision"
Link: </node?node=17>; rel="create"
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
Vary: 
X-Generator: Drupal 8 (https://www.drupal.org)
X-Drupal-Cache: MISS
Content-Type: text/html; charset=UTF-8
```
```
curl -I -u admin:islandora http://localhost:8000/node/17?_format=json
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 18:53:36 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/17>; rel="canonical"
Link: <http://localhost:8000/node/17/delete>; rel="https://drupal.org/link-relations/delete-form"
Link: <http://localhost:8000/node/17/edit>; rel="edit-form"
Link: <http://localhost:8000/node/17/revisions>; rel="version-history"
Link: <http://localhost:8000/node/17>; rel="https://drupal.org/link-relations/revision"
Link: <http://localhost:8000/node?node=17>; rel="https://drupal.org/link-relations/create"
Link: <http://localhost:8000/media/17>; rel="related"; title="Web Image"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
Vary: 
X-Generator: Drupal 8 (https://www.drupal.org)
Content-Type: application/json

```

Look OK?

**Comment by whikloj** [comment]
@mjordan no it looks like you are missing the headers. I just pulled this in and this is what I see.

**Comment by whikloj** [comment]
```
ubuntu@claw:/var/www/html/drupal/web/modules/contrib$ curl -I http://localhost:8000/node/1
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 18:55:00 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/1?_format=jsonld>; rel="alternate"; type="application/ld+json"
Link: <http://localhost:8000/node/1?_format=json>; rel="alternate"; type="application/json"
Link: </node/1>; rel="canonical"
Link: </node/1>; rel="shortlink"
Link: </node/1>; rel="revision"
Link: </node?node=1>; rel="create"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
X-Drupal-Cache: MISS
Content-Type: text/html; charset=UTF-8

ubuntu@claw:/var/www/html/drupal/web/modules/contrib$ curl -I -uadmin:islandora http://localhost:8000/node/1?_format=jsonld
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 18:55:43 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/1>; rel="canonical"
Link: <http://localhost:8000/node/1/delete>; rel="https://drupal.org/link-relations/delete-form"
Link: <http://localhost:8000/node/1/edit>; rel="edit-form"
Link: <http://localhost:8000/node/1/revisions>; rel="version-history"
Link: <http://localhost:8000/node/1>; rel="https://drupal.org/link-relations/revision"
Link: <http://localhost:8000/node?node=1>; rel="https://drupal.org/link-relations/create"
Link: <http://localhost:8000/node/1?_format=json>; rel="alternate"; type="application/json"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
Content-Type: application/ld+json

ubuntu@claw:/var/www/html/drupal/web/modules/contrib$ curl -I -uadmin:islandora http://localhost:8000/node/1?_format=json
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 18:56:07 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/1>; rel="canonical"
Link: <http://localhost:8000/node/1/delete>; rel="https://drupal.org/link-relations/delete-form"
Link: <http://localhost:8000/node/1/edit>; rel="edit-form"
Link: <http://localhost:8000/node/1/revisions>; rel="version-history"
Link: <http://localhost:8000/node/1>; rel="https://drupal.org/link-relations/revision"
Link: <http://localhost:8000/node?node=1>; rel="https://drupal.org/link-relations/create"
Link: <http://localhost:8000/node/1?_format=jsonld>; rel="alternate"; type="application/ld+json"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
Content-Type: application/json
```

**Comment by DiegoPino** [comment]
@mjordan any chance you updated recently drupal core? maybe you need to hit /update.php page?

**Comment by whikloj** [comment]
@dannylamb possibly unrelated but I just noticed that I can make an unauthenticated GET request against the REST endpoints now. 
```
> curl -I "http://localhost:8000/node/2?_format=jsonld"
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 19:00:27 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/2>; rel="canonical"
Link: <http://localhost:8000/node/2/delete>; rel="https://drupal.org/link-relations/delete-form"
Link: <http://localhost:8000/node/2/edit>; rel="edit-form"
Link: <http://localhost:8000/node/2/revisions>; rel="version-history"
Link: <http://localhost:8000/node/2>; rel="https://drupal.org/link-relations/revision"
Link: <http://localhost:8000/node?node=2>; rel="https://drupal.org/link-relations/create"
Link: <http://localhost:8000/node/2?_format=json>; rel="alternate"; type="application/json"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
X-Drupal-Cache: HIT
Content-Type: application/ld+json
```

Should I not be able to do that?

File: tests/src/Functional/NodeLinkHeaderTest.php [hunk 1]
```diff
@@ -127,12 +127,41 @@ public function testRelatedLinkHeader() {
...
+      "Node must have link header pointing to jsonld REST endpoint when visiting the json REST endpoint."
+    );
+
     // Log in as anonymous.
     $account = $this->drupalCreateUser();
     $this->drupalLogin($account);
 
     // Visit the referencer. It should return a rel="related" link header
-    // for both the referenced node and media entity.
+    // for both the referenced node bun not the media entity b/c permissions.
```
Comment (whikloj) [inline tests/src/Functional/NodeLinkHeaderTest.php]: Typo: `bun`

**Comment by mjordan** [comment]
@DiegoPino everything appears up to date.

**Comment by mjordan** [comment]
I'm on the right brach, I've cleared my cache. I copied @whikloj's commands into my terminal and ran them, but am still missing the headers. Are there any other dependencies that I need to check?

**Comment by whikloj** [comment]
@mjordan you uninstall and re-installed the Islandora module? 

Not sure if that makes any difference, but that is what I did as I have mine linked out of my vagrant box.

**Comment by mjordan** [comment]
@whikloj didn't do that. `drupal module:unistall` doesn't automagically take  care of dependent modules (in this case, islandora_collection and islandora_image).... do I need to do each one individually? Sorry, still getting my D8 sea legs here :baby:

**Comment by whikloj** [comment]
@mjordan no it doesn't I wrote these [shell scripts](https://gist.github.com/whikloj/614fa21d7145b09197fd1e8513dd2f41) to help. 

I leave them in my playbook directory and then from inside the vagrant box you can call them at `~/islandora/islandora_uninstall.sh`

**Comment by whikloj** [comment]
I should say, that helps uninstall. 

You'll still need to run `drupal module:install islandora islandora_collection islandora_image` once you have updated your module.

**Comment by Natkeeran** [comment]
Uninstalled islandora.  Got the PR.  Seems to work for me.

```
nat@natubuntu:~/Desktop/claw-playbook$ curl -I http://localhost:8000/node/1
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 20:12:16 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/1?_format=jsonld>; rel="alternate"; type="application/ld+json"
Link: <http://localhost:8000/node/1?_format=json>; rel="alternate"; type="application/json"
Link: </node/1>; rel="canonical"
Link: </node/1>; rel="shortlink"
Link: </node/1>; rel="revision"
Link: </node?node=1>; rel="create"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
X-Drupal-Cache: MISS
Content-Type: text/html; charset=UTF-8

nat@natubuntu:~/Desktop/claw-playbook$ curl -I -u admin:islandora http://localhost:8000/node/1?_format=json
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 20:13:40 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/1>; rel="canonical"
Link: <http://localhost:8000/node/1/delete>; rel="https://drupal.org/link-relations/delete-form"
Link: <http://localhost:8000/node/1/edit>; rel="edit-form"
Link: <http://localhost:8000/node/1/revisions>; rel="version-history"
Link: <http://localhost:8000/node/1>; rel="https://drupal.org/link-relations/revision"
Link: <http://localhost:8000/node?node=1>; rel="https://drupal.org/link-relations/create"
Link: <http://localhost:8000/node/1?_format=jsonld>; rel="alternate"; type="application/ld+json"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
Content-Type: application/json

nat@natubuntu:~/Desktop/claw-playbook$ curl -I -u admin:islandora http://localhost:8000/node/1?_format=jsonld
HTTP/1.1 200 OK
Date: Mon, 19 Mar 2018 20:15:41 GMT
Server: Apache/2.4.18 (Ubuntu)
X-Powered-By: PHP/7.0.28-0ubuntu0.16.04.1
Cache-Control: must-revalidate, no-cache, private
Link: <http://localhost:8000/node/1>; rel="canonical"
Link: <http://localhost:8000/node/1/delete>; rel="https://drupal.org/link-relations/delete-form"
Link: <http://localhost:8000/node/1/edit>; rel="edit-form"
Link: <http://localhost:8000/node/1/revisions>; rel="version-history"
Link: <http://localhost:8000/node/1>; rel="https://drupal.org/link-relations/revision"
Link: <http://localhost:8000/node?node=1>; rel="https://drupal.org/link-relations/create"
Link: <http://localhost:8000/node/1?_format=json>; rel="alternate"; type="application/json"
X-Drupal-Dynamic-Cache: MISS
X-UA-Compatible: IE=edge
Content-language: en
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Expires: Sun, 19 Nov 1978 05:00:00 GMT
X-Generator: Drupal 8 (https://www.drupal.org)
Content-Type: application/ld+json

nat@natubuntu:~/Desktop/claw-playbook$
```

**Comment by mjordan** [comment]
Now that @Natkeeran and @whikloj  have tested, I'm going to cease and desist..... I uninstalled islandora_image and islandora_collection and now trying islandora and I'm getting `[ERROR] Attempt to create a field field_height that does not exist on entity type media. `

**Comment by Natkeeran** [comment]
@mjordan seems like you are running into a known uninstall problem: https://github.com/Islandora-CLAW/CLAW/issues/685 related to drupal core.  At the end of the thread, there is a script to uninstall islandora by @whikloj.

**Comment by mjordan** [comment]
@Natkeeran thanks I'll give that a try later. Have spun down my claw VM to test some 7.x stuff.

**Comment by whikloj** [comment]
@mjordan as @Natkeeran said, there is something in our setup that gets stuck in Drupal. Hence the field `config:delete` commands in my script. You can just copy them and paste them directly in to you vagrant VM too.
https://gist.github.com/whikloj/614fa21d7145b09197fd1e8513dd2f41#file-islandora_uninstall-sh-L9-L12

File: tests/src/Functional/NodeLinkHeaderTest.php (hunk 1 above)
Comment (dannylamb) [reply tests/src/Functional/NodeLinkHeaderTest.php]: That's a delicious typo :bread:

**Comment by dannylamb** [comment]
@whikloj 

> @dannylamb possibly unrelated but I just noticed that I can make an unauthenticated GET request against the REST endpoints now.

You should be able to make a REST request for a node because anonymous can view content, but not against a Media (anonymous cannot view Media).  To that effect, as anonymous you should not see the link headers pointing towards Media when issuing a GET or HEAD request.

**Pull request merged**